new py-amplpy package

Sangu-Mbekelu opened the pull request:
depends on !35357

Sangu-Mbekelu committed:
new py-amplpy package

sm2939 commented:
@spackbot fix style

Sangu-Mbekelu committed:
[@spackbot] updating style on behalf of sm2939

tldahlgren reviewed (changes requested):
As with #35357 , the directory for this package needs to be moved to under `var/spack/repos/builtin/packages`.

Confirmed the sha256.

tldahlgren commented on var/spack/repos/builtin/py-amplpy/package.py:
```diff
@@ -0,0 +1,24 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
```
```suggestion
# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
```

adamjstewart commented on var/spack/repos/builtin/py-amplpy/package.py:
```diff
@@ -0,0 +1,24 @@
+    pypi = "amplpy/amplpy-0.8.6.tar.gz"
+
+    version("0.8.6", sha256="ad0945d69f75e7762802bb54849009717fbcf226a6da6f37b539d9534bdcf68d")
+
```
missing build dep on setuptools

Sangu-Mbekelu committed:
Update package.py

Sangu-Mbekelu committed:
Rename var/spack/repos/builtin/py-amplpy/package.py to var/spack/repos/builtin/packages/py-amplpy/package.py

Edited file to change copyright year/dependencies and changed the directory of the file

tldahlgren reviewed (approved):
Reconfirmed the sha256.

tldahlgren committed:
Merge branch 'develop' into sm2939/py-amplpy

tldahlgren commented:
Took the liberty of `Update branch` button to pick up the dependency package.
